data/src/main/java/me/androidbox/data/local/agenda/AgendaLocalRepositoryImp.kt

```diff
@@ -0,0 +1,94 @@
+package me.androidbox.data.local.agenda
+
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import me.androidbox.data.local.dao.EventDao
+import me.androidbox.data.local.dao.ReminderDao
+import me.androidbox.data.local.dao.TaskDao
+import me.androidbox.data.mapper.*
+import me.androidbox.domain.agenda.model.Agenda
+import me.androidbox.domain.authentication.ResponseState
+import me.androidbox.domain.authentication.remote.AgendaLocalRepository
+import me.androidbox.domain.repository.AgendaRemoteRepository
+import java.time.ZoneId
+import javax.inject.Inject
+
+class AgendaLocalRepositoryImp @Inject constructor(
+    private val eventDao: EventDao,
+    private val taskDao: TaskDao,
+    private val reminderDao: ReminderDao,
+    private val agendaRemoteRepository: AgendaRemoteRepository
+) : AgendaLocalRepository {
+
+    override fun fetchAgenda(startTimeStamp: Long, endTimeStamp: Long): Flow<ResponseState<Agenda>> {
+        return flow<ResponseState<Agenda>> {
```

Comment:
You are using Flow, yet only emiting the value just once (at the end). So you aren´t really taking advantage of Flows over a regular method

One way to take advantage would be to do:

Get items from the DB and emit them
Get items from API
Save to db
Emit those items

Most of the times, you´ll emit the same values (which if you use flow´s "Distinct", nothing´s gonna happen), but if you get stuff from API that are new, they will be saved on the db and then emitted

This is important, since right now you are calling the API, and if the API takes 5 seconds for some reason, you aren´t displaying any items to the user

The whole point of an offline first is to show the user info, even if the connectivity is slow or non existant

Reply:
@MKiperszmid @philipplackner I understand the concept that we first want to show the user the agenda items before fetching from the API. And if there is items we should insert them, and emit them again. However, I am having issues with the coding part of this using flows. That is were I need help with.

I think I am doing something wrong with my AgendaLocalRepositoryImp class when I am fetching and emitting. Please help

Reply:
I see you added the code that does this. Still need help?

Reply:
@MKiperszmid Yes, changing to suspend functions resolved that issue. But not sure why the flow didn't work. Maybe that is a questions to @philipplackner

Reply:
@steve1rm Maybe you didn't launch the flow in the ViewModel by using `collectLatest` or `launchIn`?